src/multilinear.jl

```diff
@@ -296,8 +334,74 @@ function permutation_matrix(
 ) where {T}
     dims = dims isa Integer ? fill(dims, length(perm)) : dims
     d = prod(dims)
-    id = SA.sparse(one(T)*I, (d, d))
+    id = SA.sparse(one(T) * I, (d, d))
     return permute_systems(id, perm, dims; rows_only = true)
 end
 permutation_matrix(dims, perm) = permutation_matrix(Bool, dims, perm)
 export permutation_matrix
+
+@doc """
+    trace_replace(X::AbstractMatrix, remove::AbstractVector, dims::AbstractVector = _equal_sizes(X))
+
+Takes the partial trace of matrix `X` with subsystem dimensions `dims` and replace the removed subsystems by identity.
+If the argument `dims` is omitted two equally-sized subsystems are assumed.
+""" trace_replace(X::AbstractMatrix, remove::AbstractVector, dims::AbstractVector = _equal_sizes(X))
+
+for (T, limit, wrapper) ∈
+    [(:AbstractMatrix, :dpt, :identity), (:(Hermitian), :j, :(Hermitian)), (:(Symmetric), :j, :(Symmetric))]
+    @eval begin
+        function trace_replace(
+            X::$T,
+            replace::AbstractVector{<:Integer},
+            dims::AbstractVector{<:Integer} = _equal_sizes(X)
+        )
+            isempty(replace) && return X
+            length(replace) == length(dims) && return SA.sparse(tr(X) / prod(dims[replace]) * I, size(X))
+
+            nsys = length(dims)
+            nsys_rp = length(replace)
+            nsys_kept = nsys - nsys_rp
+
+            keep = _subsystems_complement(replace, nsys)
+            ssys_step = _step_sizes_subsystems(dims)
+
+            dims_keep = dims[keep] # The tensor dimensions of Y
+            dims_rp = dims[replace] # The tensor dimensions of the traced out systems
+            ssys_step_keep = ssys_step[keep]
+            ssys_step_rp = ssys_step[replace]
+
+            step_iterator_keep = _step_iterator(dims_keep, ssys_step_keep)
+            step_iterator_rp = _step_iterator(dims_rp, ssys_step_rp)
+            step_iterator_rp .-= 1
+
+            #Take the partial trace
+            dpt = prod(dims_keep)
+            pt = partial_trace(X, replace, dims)
+            pt /= prod(dims_rp) # normalize for trace preservation
```

Comment:
```suggestion
            pt ./= prod(dims_rp) # normalize for trace preservation
```
avoid allocating a temporary matrix